pkg/rpcclient/notary/contract.go

```diff
@@ -234,3 +240,40 @@ func withdrawScript(from util.Uint160, to util.Uint160) []byte {
 	script, _ := smartcontract.CreateCallWithAssertScript(Hash, "withdraw", from.BytesBE(), to.BytesBE())
 	return script
 }
+
+// ToStackItem implements stackitem.Convertible interface.
+func (d *OnNEP17PaymentData) ToStackItem() (stackitem.Item, error) {
+	return stackitem.NewArray([]stackitem.Item{
+		stackitem.Make(d.Account),
+		stackitem.Make(d.Till),
+	}), nil
+}
+
+// FromStackItem implements stackitem.Convertible interface.
+func (d *OnNEP17PaymentData) FromStackItem(si stackitem.Item) error {
+	arr, ok := si.Value().([]stackitem.Item)
+	if !ok {
+		return fmt.Errorf("unexpected stackitem type: %s", si.Type())
+	}
+	if len(arr) != 2 {
+		return fmt.Errorf("unexpected number of fields: %d vs %d", len(arr), 2)
+	}
+
+	if arr[0] != stackitem.Item(stackitem.Null{}) {
+		accBytes, err := arr[0].TryBytes()
+		if err != nil {
+			return fmt.Errorf("failed to retrieve account bytes: %w", err)
+		}
+		acc, err := util.Uint160DecodeBytesBE(accBytes)
+		if err != nil {
+			return fmt.Errorf("failed to decode acctount bytes: %w", err)
+		}
+		d.Account = &acc
+	}
+	till, err := arr[1].TryInteger()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve till: %w", err)
+	}
+	d.Till = uint32(till.Int64())
```

Comment:
What if it's `!till.IsInt64()`? What if it's larger than `math.MaxUint32`?

Reply:
I thought about it, but Notary can't properly handle these data, thus I've disided to skip these checkc. Although you're right and these methods should work with any data.

Fixed.